src/runtime/template_set.ts

```diff
@@ -97,7 +104,8 @@ export class TemplateSet {
       this.templates[name] = function (context, parent) {
         return templates[name].call(this, context, parent);
       };
-      const template = templateFn(this, bdom, helpers);
+      const runtimeUtils = { ...helpers, __globals__: this.globalValues };
```

Comment:
this is done everytime a component is instantiated. it would be cheaper to do it just once in the setup. so, remove this line